src/frontends/pytorch/src/op/upsample.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/opsets/opset8.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+OutputVector translate_upsample2d(NodeContext& context, opset8::Interpolate::InterpolateMode interpolate_mode) {
+    auto data = context.get_input(0);
+    std::vector<size_t> pad{0};
+    auto size_mode = opset8::Interpolate::ShapeCalcMode::SIZES;
+    bool align_corners = false;
+    int scale_id = 2;
+    if (interpolate_mode == opset8::Interpolate::InterpolateMode::LINEAR_ONNX) {
+        scale_id = 3;
+        if (!context.input_is_none(2)) {
+            align_corners = context.const_input<bool>(2);
+        }
+    }
+    auto target_axes = std::make_shared<opset8::Constant>(element::i32, Shape{2}, std::vector<int>({2, 3}));
+    auto scales = context.mark_node(std::make_shared<opset8::Constant>(element::f32, Shape{2}, std::vector<double>({1, 1})));
+    auto output_sizes = std::make_shared<opset8::Constant>(element::i32, Shape{2}, std::vector<int>({1, 1}));
+    if (context.input_is_none(1)) {
+        FRONT_END_OP_CONVERSION_CHECK(!context.input_is_none(scale_id), "Scale or Output size should be provided");
+        auto spatial_scales = context.const_input<std::vector<double>>(scale_id);
```

Comment:
I think scales can be non-constant in many cases, so better would be to have sub-graph for supporting scales (with If operation I guess)

Reply:
handelered (but I find simplier solution then if, just multipy to const of required size)